pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -421,12 +421,16 @@ func (e *EndpointController) syncService(key string) error {
 	}
 
 	var tolerateUnreadyEndpoints bool
-	if v, ok := service.Annotations[TolerateUnreadyEndpointsAnnotation]; ok {
-		b, err := strconv.ParseBool(v)
-		if err == nil {
-			tolerateUnreadyEndpoints = b
-		} else {
-			utilruntime.HandleError(fmt.Errorf("Failed to parse annotation %v: %v", TolerateUnreadyEndpointsAnnotation, err))
+	if service.Spec.PublishNotReadyAddresses == true {
```

Comment:
This has to be the other way - the annotation has to be preferred if it exists.  Otherwise it is not fully backwards compatible

Reply:
@thockin thanks! I have amended it.